File: Backend/fastApiProject/model/assignment.py
from sqlalchemy import Column, Integer, Text, ForeignKey, DateTime, Enum, func
from sqlalchemy.orm import relationship

from config.database import Base
from model.enums import AssignmentStatus


class Assignment(Base):
    __tablename__ = 'assignment'

    id = Column(Integer, primary_key=True)
    description = Column(Text)
    status = Column(Enum(AssignmentStatus), default=AssignmentStatus.PENDING, nullable=False)
    assigned_to = Column(Integer, ForeignKey('users.id'), nullable=False)
    assigned_by = Column(Integer, ForeignKey('users.id'), nullable=False)
    disaster_id = Column(Integer, ForeignKey('disaster.id'), nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # Relationships
    volunteer = relationship("User", back_populates="assignments_received", foreign_keys=[assigned_to])
    coordinator = relationship("User", back_populates="assignments_given", foreign_keys=[assigned_by])
    disaster = relationship("Disaster", back_populates="assignments")

File: disaster-live-alerts/src/main/java/com/example/disaster_live_alerts/model/Assignment.java
package com.example.disaster_live_alerts.model;


import com.example.disaster_live_alerts.enums.AssignmentStatus;
import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "assignment")
public class Assignment {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "assigned_to")
    private User assignedTo;

    @Enumerated(EnumType.STRING)
    private AssignmentStatus status;

    private String description;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();
}
